### PR TITLE
11642_Trainers_CRUD

### PR DESCRIPTION
**Pull request by phongtt-1837**
https://edu-redmine.sun-asterisk.vn/issues/11642

### CODE REVIEW

**Comment by phongtt-1837** [comment]
ready

File: app/Http/Controllers/TrainerController.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $languages = $this->trainer->getLanguage();
+        $offices = $this->trainer->getOffice();
+        $trainer = $this->trainer->get([], $id);
+
+        return view('admin.trainers.edit', compact('trainer','languages','offices'));
```
Comment (BunnyPi04) [inline app/Http/Controllers/TrainerController.php]: convention

File: app/Models/User.php [hunk 2]
```diff
@@ -16,7 +16,7 @@ class User extends Authenticatable
      * @var array
      */
     protected $fillable = [
-        'name', 'email', 'password',
+        'name', 'email', 'password','roles'
```
Comment (BunnyPi04) [inline app/Models/User.php]: convention

**Comment by phongtt-1837** [comment]
ready

File: app/Http/Controllers/TrainerController.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $current_user = $this->trainer->getCurrentUser();
+        $trainers = $this->trainer->all();
+
+        return view('admin.trainers.index', compact('current_user','trainers'));
```
Comment (code1phut) [inline app/Http/Controllers/TrainerController.php]: convention

File: app/Http/Controllers/TrainerController.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $languages = $this->trainer->getLanguage();
+        $offices = $this->trainer->getOffice();
+
+        return view('admin.trainers.create', compact('languages','offices'));
```
Comment (code1phut) [inline app/Http/Controllers/TrainerController.php]: convention

File: app/Http/Controllers/TrainerController.php (hunk 1 above)
Comment (code1phut) [inline app/Http/Controllers/TrainerController.php]: convention

File: app/Models/User.php (hunk 2 above)
Comment (code1phut) [inline app/Models/User.php]: convention

File: app/Repositories/Contracts/TrainerRepository.php [hunk 5]
```diff
@@ -2,20 +2,99 @@
...
+        return $offices = Office::all();
+    }
+
+    public function store($data)
+    {
+        $user = $this->user->create([
+            'name' => $data['name'],
+            'email' => $data['email'],
+            'password' => Hash::make('abc123'),
+            'roles' => '2',
```
Comment (code1phut) [inline app/Repositories/Contracts/TrainerRepository.php]: hard code

File: app/Repositories/Contracts/TrainerRepository.php [hunk 6]
```diff
@@ -2,20 +2,99 @@
...
+use App\Models\Language;
+use App\Models\StaffType;
 use App\Repositories\TrainerRepositoryInterface;
 use App\Repositories\Contracts\BaseRepository;
 use App\Models\Trainer;
+use App\Models\User;
+use App\Models\Office;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+
```
Comment (thucpv-1748) [inline app/Repositories/Contracts/TrainerRepository.php:15]: thưa dòng trắng nhóe

File: app/Repositories/Contracts/TrainerRepository.php [hunk 7]
```diff
@@ -2,20 +2,99 @@
...
+            'phone' => $data['phone'],
+            'language_id' => $data['language_id'],
+            'user_id' => $user->id,
+            'office_id' => $data['office_id'],
+        ]);
+    }
+
+    public function get($relation = [], $id)
+    {
+        $trainer = $this->trainer::where('id', $id)->first();
```
Comment (thucpv-1748) [inline app/Repositories/Contracts/TrainerRepository.php]: trường hợp ko có ID thì sao

File: resources/views/admin/trainers/form.blade.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            <label for="complaintinput4">{{ __('Phone') }}</label>
+            <input type="text" id="phone" class="form-control round"
+                   name="phone" value="{{ isset($trainer) ? $trainer->phone : old('phone') }}">
+        </div>
+        <div class="form-group">
+            <label for="issueinput5">{{ __('Language') }}</label>
+            <select id="language_id" name="language_id" class="form-control" data-toggle="tooltip"
+                    data-trigger="hover" data-placement="top" data-title="Priority">
+                @foreach ($languages as $language)
+                    <option value="{{ $language->id }}" {{ (isset($trainer) && $trainer->language_id == $language->id) ? 'selected' : ''}}>{{ $language->name }}</option>
```
Comment (thucpv-1748) [inline resources/views/admin/trainers/form.blade.php]: convention nhé

File: resources/views/admin/trainers/form.blade.php [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                @endforeach
+            </select>
+        </div>
+        <div class="form-group">
+            <label for="issueinput5">{{ __('Office') }}</label>
+            <select id="office_id" name="office_id" class="form-control" data-toggle="tooltip"
+                    data-trigger="hover" data-placement="top" data-title="Priority">
+                @foreach ($offices as $office)
+                    <option value="{{ $office->id }}" {{ (isset($trainer) && $trainer->office_id == $office->id) ? 'selected' : ''}}>{{ $office->name }}</option>
```
Comment (thucpv-1748) [inline resources/views/admin/trainers/form.blade.php]: convention

File: resources/views/admin/trainers/index.blade.php [hunk 10]
```diff
@@ -0,0 +1,61 @@
+@extends('admin.layouts.master')
+@section('title', 'All Trainers')
```
Comment (thucpv-1748) [inline resources/views/admin/trainers/index.blade.php]: I18n

File: resources/views/admin/trainers/index.blade.php [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+                                        <th>{{ __('Language') }}</th>
+                                        <th>{{ __('Date of Birth') }}</th>
+                                        <th>{{ __('Phone') }}</th>
+                                        <th>{{ __('Address') }}</th>
+                                        <th>{{ __('Office') }}</th>
+
+                                    </tr>
+                                    </thead>
+                                    <tbody>
+                                    @foreach($trainers as $trainer)
```
Comment (thucpv-1748) [inline resources/views/admin/trainers/index.blade.php]: convention

**Comment by phongtt-1837** [comment]
ready

**Pull request merged**